### ISSUE TITLE
EPIC 2: Vehicle Image Processing Pipeline

### ISSUE THREAD

**Issue by ikabrain** [CivicPulse-TIET/civicpulse-ml#3]
- [x] Collect sample car image open datasets
- [ ] Implement pre-processing module
- [x] Integrate YOLOv8 / Detectron2
- [x] Draw bounding boxes for debugging
- [ ] Save cropped to temp
- [ ] Unit test for detection output

**Comment by ikabrain** [comment]
> [!NOTE]
>  Template for pipeline from [this vid](https://youtu.be/rysdr4khB5k?si=CGRs66OgeqV1ckwF):-
>
> 1. **Data cleaning:** NaNs, duplicates, outliers
> 2. **Data transformation & encoding:** Binary (Image) to numeric sequence, augment under-represented data
> 3. **Data preprocessing:** Normalisation, standardisation, etc
> 4. **Split into train-validation-test sets:** Stratify along target if dataset is imbalanced, assume `random_state` as a hyperparameter, use same scales for train & test
> 5. **Hyperparameter tuning:** Try tree-parzen estimators (TPE) using `optuna` on the VALIDATION set
> 6. **Select appropriate models:** Analyse performance using apprpriate metrics, see if overfitting / underfitting, CHECK STATIONARITY OF DATA
> 7. **Deploy model:** Save preprocessing weights, consider model drift for dynamic data, calculate compute time complexity

**Comment by ikabrain** [comment]
> [!NOTE]
> Use [MLFlow](https://www.youtube.com/watch?v=6ngxBkx05Fs) for tracking models & scores

**Comment by ikabrain** [comment]
Marking this issue as closed since a PR tackles the main concerns with the pipeline and fine-tuning the YOLO model weights

**Comment by ikabrain** [comment]
Issue has been reopened as testing the YOLO model is pending and needs to be continued from branch @ [ikabrain/
civicpulse-ml-dev:model/test-yolo](https://github.com/ikabrain/civicpulse-ml-dev/tree/model/test-yolo).

For now, going on an indefinite break from the project.
